### ISSUE TITLE
zenodo-bibtex.cgi doesn't return all components if some are unchanged

### ISSUE THREAD

**Issue by dorugeber** [firedrakeproject/firedrake#1157]
E.g. http://firedrakeproject.org/zenodo-bibtex.cgi?tag=Firedrake_20171018.0 doesn't have a COFFEE component.

I guess this is because Zenodo doesn't make a new record if the Github release has the same commit SHA as that of an existing Zenodo release.  The Github release is shown at https://github.com/coneoproject/COFFEE/releases, however the current commit SHA is the same as for https://zenodo.org/record/573267.

**Comment by wence-** [comment]
Yes, I observed this, it's somewhat annoying, but I'm not immediately sure what to do.  One option:

Given a tag, compute the "earliest" tag of that commit from the repos, now we have a set of canonical tags for that tag.  We can ask zenodo for the bibtex of these tags.

Make sense?
